### PR TITLE
Obsolete gateway configuration API

### PR DESCRIPTION
**Pull request by mauroservienti**

### CODE REVIEW

File: src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs [hunk 1]
```diff
@@ -27,6 +31,10 @@ public static PersistenceExtensions<RavenDBPersistence> UseDocumentStoreForGatew
         /// </summary>
         /// <param name="cfg"></param>
         /// <param name="storeCreator">A Func that will create the document store on NServiceBus initialization.</param>
+        [ObsoleteEx(
+            Message = "RavenDB gateway persistence has been moved to the NServiceBus.Gateway.RavenDB dedicated package.",
+            RemoveInVersion = "9.0.0",
+            TreatAsErrorFromVersion = "7.0.0")]
```
Comment (mauroservienti) [inline src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs]: @DavidBoike is this correct or should it be?

- TreatAsErrorFromVersion = "7.0.0"
- RemoveInVersion = "8.0.0"

Comment (DavidBoike) [reply src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs]: Currently on Major 6, so as you have it here is correct:

* Obsoletes with warn in 6
* Obsoletes with error in 7
* Removed in 9

Comment (DavidBoike) [reply src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs]: I can't count. 8 does exist. Should be RemoveInVersion = 8.

File: src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs [hunk 2]
```diff
@@ -27,6 +31,10 @@ public static PersistenceExtensions<RavenDBPersistence> UseDocumentStoreForGatew
         /// </summary>
         /// <param name="cfg"></param>
         /// <param name="storeCreator">A Func that will create the document store on NServiceBus initialization.</param>
+        [ObsoleteEx(
+            Message = "RavenDB gateway persistence has been moved to the NServiceBus.Gateway.RavenDB dedicated package.",
+            RemoveInVersion = "9.0.0",
```
Comment (DavidBoike) [inline src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs]:
```suggestion
            RemoveInVersion = "8.0.0",
```

File: src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs [hunk 3]
```diff
@@ -16,6 +16,10 @@ public static class RavenDbGatewayDeduplicationSettingsExtensions
         /// </summary>
         /// <param name="cfg"></param>
         /// <param name="documentStore">The document store to use</param>
+        [ObsoleteEx(
+            Message = "RavenDB gateway persistence has been moved to the NServiceBus.Gateway.RavenDB dedicated package.",
+            RemoveInVersion = "9.0.0",
```
Comment (DavidBoike) [inline src/NServiceBus.RavenDB/Gateway/RavenDbGatewayDeduplicationSettingsExtensions.cs]:
```suggestion
            RemoveInVersion = "8.0.0",
```

**Pull request merged**